crypto/bls_multisig.go

```diff
@@ -188,3 +192,82 @@ func VerifySignatureOneMessage(pks []PublicKey, s Signature,
 	}
 	return aggPk.Verify(s, message, kmac)
 }
+
+// BatchVerifySignaturesOneMessage is a batch verification of multiple
+// BLS signatures of a single message against multiple BLS public keys that
+// is faster than verifying the signatures one by one.
+//
+// Each signature at index (i) of the input signature slice is verified against
+// the public key of the same index (i) in the input key slice.
+// The input hasher is the same used to generate all signatures.
+// The returned boolean slice is a slice so that the value at index (i) is true
+// if signature (i) verifies against public key (i), and false otherwise.
+//
+// Membership checks are performed on the input signatures but not on the input public
+// keys (which is supposed to have happened outside this function using the library
+// key generation or bytes decode function).
+// An error is returned if the key slice is empty.
+func BatchVerifySignaturesOneMessage(pks []PublicKey, sigs []Signature,
+	message []byte, kmac hash.Hasher) ([]bool, error) {
+	_ = newBLSBLS12381()
+
+	// public keys check
+	if len(pks) == 0 || len(pks) != len(sigs) {
+		return []bool{}, fmt.Errorf("key list length is not valid")
+	}
+
+	verifBool := make([]bool, len(sigs))
+	// hasher check
+	if kmac == nil {
+		return verifBool, errors.New("VerifyBytes requires a Hasher")
+	}
+
+	if kmac.Size() < opSwUInputLenBLSBLS12381 {
+		return verifBool, fmt.Errorf("hasher with at least %d output byte size is required, current size is %d",
```

Comment:
so you mentioned you changed the other implementation to match this previously. I'm guessing it is rather more important to keep the error here?

Reply:
You are right, I made more changes on this branch after that discussion to make sure signatures with wrong lengths do not cause crashes. 

In this case, the condition is made on the hasher not the input signature, so it makes sense to keep the error. It only means that the verifier is using the wrong hasher.